src/function/gds/awsp_paths.cpp

```diff
@@ -26,6 +26,10 @@ class AWSPPathsEdgeCompute : public EdgeCompute {
         bool fwdEdge) override {
         std::vector<nodeID_t> result;
         chunk.forEach<T>([&](auto nbrNodeID, auto edgeID, auto weight) {
+            if (weight < 0) {
+                [[unlikely]] throw RuntimeException(
```

Comment:
Can you say a bit more in the error message?
"Found negative weight {}." -> "Found negative weight {}, which is not supported yet."